stdlib/linecache.pyi

```diff
@@ -1,10 +1,16 @@
-from typing import Any, Dict, List, Optional
+from typing import Any, Callable, Protocol
 
-_ModuleGlobals = Dict[str, Any]
+_ModuleGlobals = dict[str, Any]
+_ModuleMetadata = tuple[int, float, list[str], str]
 
-def getline(filename: str, lineno: int, module_globals: Optional[_ModuleGlobals] = ...) -> str: ...
+class _SourceLoader(Protocol):
```

Comment:
This may be an overkill, but my initial attempt with `_SourceLoader = Callable[[], str | None]` has failed `stubtest_stdlib` with a message
```
stdlib/linecache.pyi:5: error: Invalid type alias: expression is not a valid type
stdlib/linecache.pyi:5: error: Unsupported left operand type for | ("Type[str]")
stdlib/linecache.pyi:7: error: Variable "linecache._SourceLoader" is not valid as a type
```
A viable alternative could also be using `Optional`: `_SourceLoader = Callable[[], Optional[str]]`.